handlers/module_image.go

```diff
@@ -459,6 +459,12 @@ func (img *ModuleImage) ReadHeader(r io.Reader, path string, version int, augmen
 		if err != nil {
 			return errors.Wrap(err, "error reading type-info")
 		}
+		if img.typeInfoV3 == nil || img.typeInfoV3.Type != img.updateType {
+			return errors.New("Type in type-info header does not match header-info: " +
+				"Corrupt Artifact. " +
+				"This was a known bug in some versions of mender-artifact prior to 3.5.1. " +
```

Comment:
Suggestion: How about: `is a known bug?`

Reply:
Good point, I will change that.